src/actions.js

```diff
@@ -44,3 +46,6 @@ export const initializeCurrentLocation = (location: Location) => ({
   type: LOCATION_CHANGED,
   payload: location
 });
+
+export const block = (callback: Function) => ({ type: BLOCK, callback });
```

Comment:
@scarney81 What were you thinking? The callback that's passed in could be anything from within the `ui` and it can either return a boolean, string, or open a slideout, so not sure how to flow type that. Curious if you got suggestions!